torch/distributed/rpc/__init__.py

```diff
@@ -162,11 +165,14 @@ def init_rpc(
         # This rendezvous state sometimes is destroyed before all processes
         # finishing handshaking. To avoid that issue, we make it global to
         # keep it alive.
-        global rendezvous_iterator
-        rendezvous_iterator = dist.rendezvous(
-            rpc_backend_options.init_method, rank=rank, world_size=world_size
-        )
-        store, _, _ = next(rendezvous_iterator)
+        if world_size:
+            global rendezvous_iterator
+            rendezvous_iterator = dist.rendezvous(
+                rpc_backend_options.init_method, rank=rank, world_size=world_size
+            )
+            store, _, _ = next(rendezvous_iterator)
+        else:
+            store = _create_store_for_rpc(rpc_backend_options, rank)
```

Comment:
This looks a bit inconsistent with the `if` clause. Does it make sense to call it `dist.join()` or `dist.connect()`?

And looks like `_create_store_for_rpc` does not seem it contains RPC specific code. If that's the case, we can keep it in `rendezvous.py` for now and also pass `rpc_backend_options.init_method` and timeout explicitly to it.

Reply:
moved it into rendezvous and creates the store based on backend_options